include/deal.II/lac/communication_pattern_base.h

```diff
@@ -17,9 +17,6 @@
 #define dealii__communication_pattern_base_h
 
 #include <deal.II/base/config.h>
-
-#ifdef DEAL_II_WITH_MPI
-
```

Comment:
We have typedefs for those basic structures in case no MPI is present to avoid too many ifdefs:
```
typedef int MPI_Comm;
const int MPI_COMM_SELF = 0;
typedef int MPI_Datatype;
typedef int MPI_Op;
```